src/nvim/edit.c

```diff
@@ -3874,7 +3872,7 @@ static int ins_compl_get_exp(pos_T *ini)
                 /* copy as much as possible of the new word */
                 if (tmp_ptr - ptr >= IOSIZE - len)
                   tmp_ptr = ptr + IOSIZE - len - 1;
-                STRNCPY(IObuff + len, ptr, tmp_ptr - ptr);
+                memcpy(IObuff + len, ptr, tmp_ptr - ptr);
```

Comment:
`IObuff` is NUL-terminated a few lines below. 

All of the `memcpy()` cases meet one of these scenarios:

- the target buffer is NUL-terminated shortly after the memcpy
- the target buffer is treated like a string shortly after the memcpy (so it must be NUL-terminated)
- the length argument was calculated from the source (which was treated like a string, so it must be NUL-terminated)